src/TokenizedStrategy.sol

```diff
@@ -1982,39 +2012,16 @@ contract TokenizedStrategy {
      * @notice Returns the domain separator used in the encoding of the signature
      * for {permit}, as defined by {EIP712}.
      *
-     * @dev This checks that the current chain id is the same as when the contract
-     * was deployed to prevent replay attacks. If false it will calculate a new
-     * domain separator based on the new chain id.
-     *
      * @return . The domain separator that will be used for any {permit} calls.
      */
     function DOMAIN_SEPARATOR() public view returns (bytes32) {
-        StrategyData storage S = _strategyStorage();
-        return
-            block.chainid == S.INITIAL_CHAIN_ID
-                ? S.INITIAL_DOMAIN_SEPARATOR
-                : _computeDomainSeparator(S);
-    }
-
-    /**
-     * @dev Calculates and returns the domain separator to be used in any
-     * permit functions for the strategies {permit} calls.
-     *
-     * This will be used at the initialization of each new strategies storage.
-     * It would then be used in the future in the case of any forks in which
-     * the current chain id is not the same as the original.
-     *
-     */
-    function _computeDomainSeparator(
-        StrategyData storage S
-    ) internal view returns (bytes32) {
         return
             keccak256(
                 abi.encode(
                     keccak256(
                         "EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"
                     ),
-                    keccak256(bytes(S.name)),
+                    keccak256(bytes("Yearn Vault")),
                     keccak256(bytes(API_VERSION)),
```

Comment:
I had assumed that the compiler is smart enough to store the individual hashes as constants in the bytecode even without declaring them as constants so the more readable version is preferred. Though I havent verified it is and will double check to make sure

We are removing the cached domain_seperator since it cannot be stored as immutables and have to be kept in storage which is more expensive to pull from rather than recalculating it every call.

Reply:
Maybe newer versions of the compiler are smarter but this one still leaves room for gas savings:

```logs
testFuzz_permit(uint256) (gas: -114 (-0.133%))
test_permit_withExpiry() (gas: -171 (-0.183%))
test_permit_badV() (gas: -14535 (-0.185%))
test_permit_replay() (gas: -171 (-0.193%))
test_permit_badS() (gas: -114 (-0.203%))
test_permit_ownerSignerMismatch() (gas: -114 (-0.203%))
test_permit_earlyNonce() (gas: -114 (-0.204%))
test_permit_differentSpender() (gas: -114 (-0.204%))
test_permit_zeroAddress() (gas: -114 (-0.204%))
testFuzz_permit_multiple(bytes32) (gas: -1140 (-0.432%))
```

Reply:
So it appears that all of those savings are actually on coming from the _name_hash.

The other two do seem to be storing the hashes as constants but the name hash is not because it is casting the string to bytes() first before hashing it which i guess is not stored in its final form

So if we dont cast the string to bytes() which we no longer need to do cause its not pulling from storage, then it appears that the cost for permit is the same as declaring them constants outside the function

Reply:
https://github.com/yearn/tokenized-strategy/pull/98/commits/773af255606dc78b6f69d757c3e41de10dd5e3d9